### PR TITLE
Treat sdr-deploy like other Ruby projects

### PR DESCRIPTION
**Pull request by mjgiarlo**
Connects to #57

## Why was this change made?

This commit spruces up sdr-deploy in a number of ways:
* Disentangle concerns (deployment, cocina-models version check, SSH connection verification)
* Use platform-agnostic method of detecting cocina-models version (`grep -e` not good on Linux)
* Stop requiring the cocina check operation to specify an environment which is not actually used in said check
* Use thor to handle CLI option parsing
* Make some commands less spammy if they contain no errors, making it easier to notice said errors
* Use bundler to declare dependencies (just like all our other Ruby projects)
* Use ruby config gem to store configuration (just like all our other Ruby projects)
* Add rubocop (to keep us honest and consistent, and to enable CI in a follow-up commit)

## How was this change tested?

Ran rubocop and ran the three commands, and they worked. Would be good to get someone else to test-drive as well.

## Which documentation and/or configurations were updated?

README, and added config.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,20 +10,20 @@ Make sure that:
...
 * You have added the public SSH key, often `~/.ssh/id_rsa.pub`, from your machine to [GitHub](https://github.com/settings/keys)
 * You have previously `ssh`-ed into all servers.
-  * NOTE: If you are unsure about this, run `./deploy.rb [qa|stage|prod] --checkssh` and watch the output for any errors!
+  * NOTE: If you are unsure about this, run `thor check_ssh -e [qa|stage|prod]` and watch the output for any errors!
 
 ### Check your ssh connection to all servers
 
 ```
-./deploy.rb [qa|stage|prod] --checkssh
+thor check_ssh -e [qa|stage|prod]
```
Comment (jcoyne) [inline README.md:18]: How about we keep `deploy` and do like sinopia_cli does, and invoke Thor: https://github.com/LD4P/sinopia_cli/blob/main/exe/sinopia-cli & https://github.com/LD4P/sinopia_cli/blob/main/lib/sinopia_cli.rb#L8

Comment (mjgiarlo) [reply README.md:18]: @jcoyne are you suggesting we have one `exe` per operation (`deploy`, `check_cocina`, `check_ssh`) or that we route all three through `deploy`?

Comment (mjgiarlo) [reply README.md:18]: @jcoyne I ask bc `deploy check_cocina` realllllly doesn't do it for me. (Checking cocina is a local operation.)

Comment (mjgiarlo) [reply README.md]: @jcoyne Made a related change. Have another look?

File: README.md [hunk 2]
```diff
@@ -1,44 +1,45 @@
...
 ```
 
 NOTE: Watch the output for any errors
 
-### Check versions of cocina
+### Check versions of cocina-models
 
 ```
-./deploy.rb [qa|stage|prod] --check-cocina
+bin/check_cocina
```
Comment (mjgiarlo) [inline README.md:27]: Test to see what happens when these commands are invoked via bx and tweak docs if necessary

File: lib/repo_updater.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+
+# Update locally cached git repositories
+class RepoUpdater
```
Comment (mjgiarlo) [inline lib/repo_updater.rb:6]: Invoke repoupdater before checking cocina

**Review by ndushay — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (mjgiarlo) [reply README.md:27]: `bundle exec` does not affect the behavior.

File: .rubocop_todo.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+# Offense count: 1
+# Configuration parameters: IgnoredMethods.
+Metrics/CyclomaticComplexity:
+  Max: 10
+
+# Offense count: 4
+# Configuration parameters: CountComments, CountAsOne, ExcludedMethods, IgnoredMethods.
+Metrics/MethodLength:
+  Max: 47
```
Comment (jcoyne) [inline .rubocop_todo.yml:22]: I believe this is now out of date.

**Comment by jcoyne** [comment]
This looks good now. The only thing I'd suggest (and feel free to disregard), is to extract a method/methods like `switch_into_project_dir() do ... end` and/or `within_each_project_dir() do ... end`

File: .rubocop_todo.yml (hunk 4 above)
Comment (mjgiarlo) [reply .rubocop_todo.yml]: Updated. thanks!

**Comment by mjgiarlo** [comment]
> This looks good now. The only thing I'd suggest (and feel free to disregard), is to extract a method/methods like `switch_into_project_dir() do ... end` and/or `within_each_project_dir() do ... end`
> 
> something like:
> 
> ```ruby
> def switch_into_project_dir(dir)
>     Bundler.with_unbundled_env do
>       Dir.chdir(dir) do
>          yield
>        end
>      end
>    end
> ```

Added. Thanks!

**Review by jcoyne — APPROVED** [review]

**Pull request merged**